code/game/objects/items/weapons/cigs.dm

```diff
@@ -103,6 +90,8 @@ LIGHTERS ARE IN LIGHTERS.DM
 				explosion(user.loc, -1, 0, 2, 3, 0, flame_range = 2)
 			F.charges--
 
+	else if(is_hot(W) > 600)
```

Comment:
I think I'll refactor `is_hot` after this PR to just be object flags; that way it's nice and simple